mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,443 @@
+            return findMostSigEffect(  effect for transcript in d['transcriptEffects'] for effect in transcript['effects']  )
+
+        data_bins = []
+        data_weights = []
+
+        for annot in annot_list:
+            curr = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).filter(
+                lambda x: x['attributes'][pop] != ".").map(
+                lambda d: (get_effect(d), math.floor(float(d['attributes'][pop])*10000)/10000)).filter(
+                lambda f: f[0] == annot).countByValue()
+            currlist = curr.items()
+            currbins = [ item[0][1] for item in currlist ]
+            currweights = [ item[1] for item in currlist ]
+            data_bins.append(currbins)
+            data_weights.append(currweights)
+
+        self.data_bins = data_bins
+        self.data_weights = data_weights
+        self.annot_list = annot_list
+        self.bins = bins
+        self.sc = ss.sparkContext
+
+    def plot(self, testMode = False):
+        if (not testMode):
+            plt.rcdefaults()
+            plt.title("Allele Frequency Distribution by Functional Category")
+            plt.hist(self.data_bins, weights=self.data_weights, bins=self.bins, normed=1, histtype = 'step', label=self.annot_list)
+            plt.legend()
+            plt.yscale('log')
+            plt.xlabel("Allele Frequency")
+            plt.ylabel("Proportion of Variants (normalized)")
+            plt.show()
+        return ( sorted(self.data_bins[0])[0:20], sorted(self.data_weights[0][0:20]) )
+
+
+class VariantDistribByPop(object):
+    """
+    Plot Allele Frequency Distribution by Population
+    """
+    def __init__(self, ss, variantRDD, pop_list=[ 'AF_NFE', 'AF_AFR'], bins=[.00001,.0001,.001,.01,.025,.05,.075,.1,.15,.2,.25,.3,.35,.4,.45,.5]):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics VariantRDD
+        :param pop_list: list of populations, defaults to [ 'AF_NFE', 'AF_AFR']
+        :param bins: integer number of bins in Histogram, or a list of bin boundaries
+        """
+
+        def getMostSigEffect(dd1):
+           """
+           Return most significant effect from list of transcript annotations
+           """
+           effectList = [ effect for transcript in dd1['transcriptEffects'] for effect in transcript['effects'] ]
+           mostSigEffect = findMostSigEffect(effectList)
+           return mostSigEffect
+
+        data_bins = []
+        data_weights = []
+
+        def getmaf(x):
+            if(x<0.5):
+                return x
+            else:
+              return 1 - x
+
+        for pop in pop_list:
+           curr = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).filter(
+             lambda x: x['attributes'][pop] != ".").map(
+             lambda d: (getMostSigEffect(d), math.floor( getmaf(float(d['attributes'][pop])) *100000)/100000)).filter(lambda f: f[1] > 0).countByValue()
```

Comment:
The divide and then multiple is being used with floor to round down.

Reply:
Why can't we just call math.floor?

Reply:
The goal of the math here is to bin data at a specific precision, for example:
0.00035 * 100000 = 3.5
floor(3.5) = 3
3*/10000 = .00003
which is what we cant because we wanted everything from 0.00003 to 0.0000399999 to become 0.00003 
Am I missing a better way to do this with only floor?